web/app/js/components/TopModule.jsx

```diff
@@ -54,6 +54,9 @@ class TopModule extends React.Component {
 
   componentDidUpdate(prevProps) {
     if (this.props.startTap && !prevProps.startTap) {
+      this.tapResultsById = {};
```

Comment:
sweeeet! thanks for fixing these.

I think it might be a little easier to understand if these were placed in `startTapStreaming` since we probably want to clear our stored indexes before starting a tap anyways. (it's true that the start tap on componentDidMount will start from a clean state because of the initializations done in the constructor, but I do think it makes sense as part of that method).

Reply:
Had it there first and then thought - why do it twice. But you are correct that this probably a bit easier to reason about. Moved it!